app/pages/lab/workflow.cjsx

```diff
@@ -53,6 +54,11 @@ EditWorkflowPage = React.createClass
     deletionError: null
     workflowCreationInProgress: false
 
+  componentWillReceiveProps: (nextProps) ->
+    if nextProps.workflow isnt @props.workflow
+      ReactDOM.findDOMNode(@refs.defaultWorkflow).defaultChecked = nextProps.project.configuration.default_workflow is nextProps.workflow.id
```

Comment:
Okay, yeah, I'm seeing that now. I think fixing that is just a matter of using `checked` instead of `defaultChecked`. The `render` method is looking at the correct resources, but using `defaultChecked` makes the input [uncontrolled](https://facebook.github.io/react/docs/forms.html), which means it's not bound to anything and won't change to match new data.